pkg/apiserver/certificate/certificate.go

```diff
@@ -110,5 +103,53 @@ func ApplyServerCert(selfSignedCert bool, client kubernetes.Interface, aggregato
 	}
 
 	caCertController := newCACertController(caContentProvider, client, aggregatorClient)
+
+	if selfSignedCert {
+		go RotateSelfSignedCertificates(context.Background(), caCertController, secureServing, rotateInterval)
```

Comment:
I have an idea, let me know what you think: instead of hardcoding when we need to do the rotation, could we inspect the generated X509 certificate, looks at the `NotAfter` field, and schedule the rotation for one hour before that?